core/types/celo_additions.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+func (h *Header) IstanbulExtra() (*IstanbulExtra, error) {
+	if h.deserializedExtra == nil {
+		extra, err := ExtractIstanbulExtra(h)
+		if err != nil {
+			return nil, err
+		}
+		h.deserializedExtra = extra
+	}
+
+	return h.deserializedExtra, nil
```

Comment:
Mh, in the worst case it get's decoded twice. Do you think the lock it worth the price for avoiding this?

Reply:
Without a lock its undefined when a go-routine will see changes made from other go routines, so in the worst case it will get decoded on every call. Also this code would still trigger the race detector.